components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,311 @@
+import org.wso2.carbon.identity.action.management.exception.ActionMgtException;
+import org.wso2.carbon.identity.action.management.internal.ActionMgtServiceComponentHolder;
+import org.wso2.carbon.identity.action.management.model.Action;
+import org.wso2.carbon.identity.action.management.model.AuthProperty;
+import org.wso2.carbon.identity.action.management.model.AuthType;
+import org.wso2.carbon.identity.action.management.model.EndpointConfig;
+import org.wso2.carbon.identity.common.testng.WithH2Database;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.secret.mgt.core.SecretManagerImpl;
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretType;
+
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+@WithH2Database(files = {"dbscripts/h2.sql"})
+public class ActionManagementDAOImplTest {
+
+    private ActionManagementDAOImpl daoImpl;
+    private static final Map<String, BasicDataSource> dataSourceMap = new HashMap<>();
+    private MockedStatic<IdentityDatabaseUtil> identityDatabaseUtil;
+    private MockedStatic<IdentityTenantUtil> identityTenantUtil;
+    private MockedStatic<IdentityUtil> identityUtil;
+    private static final String DB_NAME = "action_mgt";
+    public static final String PRE_ISSUE_ACCESS_TOKEN = "PRE_ISSUE_ACCESS_TOKEN";
+    private static final int TENANT_ID = 2;
+    private ActionSecretProcessor actionSecretProcessor;
+    private SecretManagerImpl secretManager;
+
+    private final AuthType authType = new AuthType.AuthTypeBuilder()
+            .type(AuthType.AuthenticationType.BEARER)
+            .properties(Arrays.asList(
+                    new AuthProperty.AuthPropertyBuilder()
+                            .name("accesstoken")
+                            .value("1c4a0d92-a6f5-4eed-9465-51ab1174a90f")
+                            .isConfidential(true)
+                            .build()
+            ))
+            .build();
+
+    @BeforeClass
+    public void setUpClass() throws Exception{
+
+        daoImpl = new ActionManagementDAOImpl();
+        initiateH2Database(getFilePath());
+    }
+
+    @BeforeMethod
+    public void setUp() {
+
+        identityTenantUtil = mockStatic(IdentityTenantUtil.class);
+        identityDatabaseUtil = mockStatic(IdentityDatabaseUtil.class);
+        identityUtil = mockStatic(IdentityUtil.class);
```

Comment:
Is this mocked class used anywhere ?

Reply:
identityUtil is not needed, it is removed now.